ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -2058,3 +2063,97 @@ - (BOOL)isEqual:(id)object {
 }
 
 @end
+
+#pragma mark - ORKLocationAnswerFormat
+
+@implementation ORKLocationAnswerFormat
+
+- (Class)questionResultClass {
+    return [ORKLocationQuestionResult class];
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+    if (self) {
+        
+    }
+    return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)aCoder {
+    [super encodeWithCoder:aCoder];
+}
+
++ (BOOL)supportsSecureCoding {
+    return YES;
+}
+
+- (void)validateParameters {
+    [super validateParameters];
+}
+
+- (BOOL)isAnswerValidWithString:(nullable NSString *)text {
+    BOOL isValid = [super isAnswerValidWithString:text];
+    
+    NSArray *components = [text componentsSeparatedByString:@", "];
+    if (components.count != 2) {
+        isValid = false;
+    } else {
+        NSNumberFormatter *decimalFormatter = [[NSNumberFormatter alloc] init];
+        decimalFormatter.numberStyle = NSNumberFormatterDecimalStyle;
+        
+        NSNumber *latitude = [decimalFormatter numberFromString:components[0]];
+        NSNumber *longitude = [decimalFormatter numberFromString:components[1]];
+        
+        if (latitude.doubleValue < -180.0 || latitude.doubleValue > 180.0) {
+            isValid = false;
+        } else if (longitude.doubleValue < -180.0 || longitude.doubleValue > 180.0) {
+            isValid = false;
+        }
+    }
+    
+    return isValid;
+}
+
+- (NSString *)localizedInvalidValueStringWithAnswerString:(NSString *)text {
+    if (![text length]) {
+        return nil;
+    }
+    NSDecimalNumber *num = [NSDecimalNumber decimalNumberWithString:text locale:[NSLocale currentLocale]];
+    if (!num) {
+        return nil;
+    }
+    NSString *string = nil;
+    NSNumberFormatter *formatter = [[NSNumberFormatter alloc] init];
+    formatter.numberStyle = NSNumberFormatterDecimalStyle;
+    
+    NSArray *components = [text componentsSeparatedByString:@", "];
+    if (components.count != 2) {
+        string = [NSString stringWithFormat:ORKLocalizedString(@"RANGE_ALERT_MESSAGE_OTHER", nil), text];
+    } else {
+        NSNumberFormatter *decimalFormatter = [[NSNumberFormatter alloc] init];
+        decimalFormatter.numberStyle = NSNumberFormatterDecimalStyle;
+        
+        NSNumber *latitude = [decimalFormatter numberFromString:components[0]];
+        NSNumber *longitude = [decimalFormatter numberFromString:components[1]];
```

Comment:
If we do decide that we need a translator to convert the location into a string, then this should be dedicated to a helper method, instead of re-writing this code again and again.